draft-meynell-panrg-scion-research-questions.md

```diff
@@ -238,15 +238,47 @@ One solution could be to add a “confirm full path”-flag to certain segments.
 
 ## DRKey
-* Is forward-secrecy DRKey useful and should we develop it?
-* What are the properties of the control-plane?
-  * Do we want to have any authorization of the data-plane transit undertaken at this stage?
-  * Would this obsolete firewalls?
+DRKey is a key distribution system that scales well with the number of endpoints in the network.
+It relies on two things:
+
+* Two sides of a key: A fast side, and a slow side. Sometimes called fast and slow side of the derivation.
+  The ability of deriving a key very quickly on the fast side is necessary for most of its use cases.
+* A grouping of endpoints (such as ASes): The pieces necessary to derive a key, namely the L1 keys,
+  are communicated to each keystore at each grouping (e.g., a keystore per AS).
+
+The questions related to DRKey are the following ones (not comprehensive):
+
+* Do we want to have any possible authorization that is at the moment carried out at the data-plane,
+  be moved to the control-plane? This could include e.g. authorization to deliver traffic depending on the source,
+  but also things like port numbers/ranges per source, etc.
+  * Could this obsolete firewalls? What else would be necessary?
   * What do we mean when we say "authorize transit"?
+* Could perfect forward secrecy DRKey be useful, and should we research it?
+  * What would be the trust model? Do end-users trust their ASes to ephemerally create personal keys?
+  * What would be the attacker model?
+  * Which use cases are relevant?
 
 For more info: {{I-D.garciapardo-drkey}}.
 
 ## SCMP Authentication
+In SCION, we would like to have SCMP (SCION Control Message Protocol) include authentication for
+some of the message types, e.g. the interface down type, as it would affect the path choices that the
+endpoint, and even the source AS, can make.
+
+We propose to use DRKey as the mechanism to use to derive the authentication key,
+where the fast path would be on the infrastructure side (e.g. the border router in the case of an
+interface down type of message), and the slow side being on the intended endpoint destination
+for that SCMP message (e.g. the endpoint receiving the SCMP interface down message).
+However, we have identified a number of possible issues (not comprehensive):
+
+* Denial of Service/Capability Attacks: If an endpoint receives (too) many SCMP messages,
+  it will need (too) many resources just to authenticate their origin.
+  Most of these messages could just be sent to the endpoint to exhaust its processing capacity.
+
+* Sending an SCMP message in certain cases might be an amplification factor:
+  If a border router sends an SCMP message (e.g. interface down) on all cases, even with small packets,
+  there is the risk of having that border router sending a lot of traffic to
+  a possibly unintended recipient, e.g. when the packet is not source validated.
```

Comment:
Sorry, I just saw it. How about we summarize this paragraph into: 
How shall the network provide trustworthy signals, without introducing new attack vectors due to message authentication?